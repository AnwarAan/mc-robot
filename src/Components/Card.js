const Card = ({ name, email }) => {
  return (
    <div className="tc grow bg-light-green br3 pa3 ma2 bw2 dib shadow-5 ">
      <img src={`https://robohash.org/${name}?size=200x200`} alt="rbt-img"></img>
      <p>{name}</p>
      <p>{email}</p>
    </div>
  );
};

export default Card;
