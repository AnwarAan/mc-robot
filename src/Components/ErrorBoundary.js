import { useEffect, useState } from "react";

const ErrorBoundary = (props) => {
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setHasError(false);
  });

  if (hasError === true) {
    return (
      <div>
        <h1>Something Error...</h1>
      </div>
    );
  } else {
    return <div>{props.children}</div>;
  }
};

export default ErrorBoundary;
