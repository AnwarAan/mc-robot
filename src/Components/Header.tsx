import { PureComponent } from "react";

class Header extends PureComponent {
  render() {
    return (
      <div>
        <h1>SERAPHIM</h1>
      </div>
    );
  }
}

export default Header;
