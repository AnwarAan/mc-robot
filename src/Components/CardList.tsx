import Card from "./Card";
import { IRobots } from "../Containers/App";

const CardList = ({ robots }: { robots: Array<IRobots> }) => {
  return (
    <div>
      {robots.map((robot) => (
        <Card key={robot.id} name={robot.name} email={robot.email} />
      ))}
    </div>
  );
};

export default CardList;
