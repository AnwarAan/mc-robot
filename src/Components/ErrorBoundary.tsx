import * as React from "react";
import { useEffect, useState } from "react";
import { IProps } from "../Containers/App";

const ErrorBoundary = (props: IProps) => {
  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    setHasError(false);
  }, []);

  if (hasError === true) {
    return (
      <div>
        <h1>Something Error...</h1>
      </div>
    );
  } else {
    return <div>{props.children}</div>;
  }
};

export default ErrorBoundary;
