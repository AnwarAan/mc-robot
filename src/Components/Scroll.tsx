import { IProps } from "../Containers/App";

const Scroll = (props: IProps) => {
  return <div style={{ overflow: "scroll", height: "800px" }}>{props.children}</div>;
};

export default Scroll;
