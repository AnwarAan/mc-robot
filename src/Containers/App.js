import React from "react";
import { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  getRobotPendingReducer,
  getRobotFailReducer,
  getRobotSuccessReducer,
  searchRobotdReducer,
} from "../redux/reducer";
import "./App.css";
import CardList from "../Components/CardList";
import Header from "../Components/Header";
import SearchBox from "../Components/SeacrhBox";
import ErrorBoundary from "../Components/ErrorBoundary";
import Scroll from "../Components/Scroll";
import Loading from "../Components/Loading";

function App() {
  const dispatch = useDispatch();
  const reducer = useSelector((state) => state);
  const { robots, isPending, searchField } = reducer;

  useEffect(() => {
    dispatch(getRobotPendingReducer(true));
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then((data) => dispatch(getRobotSuccessReducer(data)))
      .catch((error) => dispatch(getRobotFailReducer(error)));
  }, []);

  const onSearchChange = (event) => {
    dispatch(searchRobotdReducer(event.target.value));
  };

  const filterRobots = robots.filter((robot) => {
    return robot.name.toLowerCase().includes(searchField);
  });

  return (
    <div className="App">
      <Header />
      <SearchBox searchChange={onSearchChange} />
      <Scroll>
        {isPending ? (
          <Loading />
        ) : (
          <ErrorBoundary>
            <CardList robots={filterRobots} />
          </ErrorBoundary>
        )}
      </Scroll>
    </div>
  );
}

export default App;
