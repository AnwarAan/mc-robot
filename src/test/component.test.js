import { configure, shallow } from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";
import Card from "../Components/Card";
import CardList from "../Components/CardList";
import ErrorBoundary from "../Components/ErrorBoundary";
import Header from "../Components/Header";
import Loading from "../Components/Loading";
import Scroll from "../Components/Scroll";
import SearchBox from "../Components/SeacrhBox";
import App from "../Containers/App";

configure({ adapter: new Adapter() });

describe("test component", () => {
  it("Card", () => {
    expect(shallow(<Card />)).toMatchSnapshot();
  });

  it("Card List", () => {
    const mockRobots = [];
    expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
  });

  it("Error Boundary", () => {
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
  });

  it("Header", () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  });

  it("Loading", () => {
    expect(shallow(<Loading />)).toMatchSnapshot();
  });

  it("Scroll", () => {
    expect(shallow(<Scroll />)).toMatchSnapshot();
  });

  it("Search Box", () => {
    expect(shallow(<SearchBox />)).toMatchSnapshot();
  });
});

describe("test app", () => {
  it("app", () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
