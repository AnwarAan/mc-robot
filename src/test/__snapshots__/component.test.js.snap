// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Card 1`] = `ShallowWrapper {}`;

exports[`test component Card List 1`] = `ShallowWrapper {}`;

exports[`test component Error Boundary 1`] = `ShallowWrapper {}`;

exports[`test component Header 1`] = `ShallowWrapper {}`;

exports[`test component Loading 1`] = `ShallowWrapper {}`;

exports[`test component Scroll 1`] = `ShallowWrapper {}`;

exports[`test component Search Box 1`] = `ShallowWrapper {}`;
